File: banchus-app/src/main/java/pe/nanamochi/banchus/startup/DataDirectoryInitializer.java
package pe.nanamochi.banchus.startup;

import jakarta.annotation.PostConstruct;
import java.io.IOException;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;
import pe.nanamochi.banchus.service.StorageService;

@Slf4j
@Component
@RequiredArgsConstructor
public class DataDirectoryInitializer {
  private final StorageService storageService;

  @PostConstruct
  public void init() {
    try {
      log.info("Initializing storage infrastructure...");
      storageService.initStorage();
      log.info("Data directories initialized successfully.");
    } catch (IOException e) {
      log.error("Could not initialize data directories: {}", e.getMessage());
      throw new RuntimeException("FileSystem initialization failed", e);
    }
  }
}

File: banchus-app/src/main/kotlin/pe/nanamochi/banchus/startup/DataDirectoryInitializer.kt
package pe.nanamochi.banchus.startup

import org.slf4j.LoggerFactory
import org.springframework.boot.CommandLineRunner
import org.springframework.stereotype.Component
import pe.nanamochi.banchus.service.StorageService

@Component
class DataDirectoryInitializer(private val storageService: StorageService) : CommandLineRunner {
    private val log = LoggerFactory.getLogger(javaClass)

    override fun run(vararg args: String) {
        runCatching {
                log.info("Initializing storage infrastructure...")
                storageService.initStorage()
            }
            .onSuccess { log.info("Data directories initialized successfully.") }
            .onFailure { e ->
                log.error("Could not initialize data directories: ${e.message}")
                throw IllegalStateException("FileSystem initialization failed", e)
            }
    }
}
